src/panels/calendar/dialog-calendar-event-editor.ts

```diff
@@ -256,9 +257,28 @@ class DialogCalendarEventEditor extends LitElement {
   }
 
   private _startDateChanged(ev: CustomEvent) {
+    // Store previous event duration
```

Comment:
Not sure if you thought about this, but given that javascript dates are pretty painful you may also consider `differenceInMinutes` from `date-fns` to get back a `Duration` which may be easier to work with, (e.g. `addMinutes` to compute the new date).